Source/WebCore/platform/graphics/FontCache.cpp

```diff
@@ -128,19 +129,32 @@ struct FontCache::FontDataCaches {
 #endif
 };
 
+static FontCache*& fontCacheForCurrentThread()
+{
+    static ThreadSpecific<FontCache*>* fontCache;
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [] {
+        fontCache = new ThreadSpecific<FontCache*>;
+    });
+    return **fontCache;
+}
+
 FontCache& FontCache::forCurrentThread()
 {
-    return threadGlobalData().fontCache();
+    auto& fontCache = fontCacheForCurrentThread();
+    if (UNLIKELY(!fontCache))
+        fontCache = new FontCache;
```

Comment:
Isn't there a potential memory leak if we do not delete fontCache in FontCache::destroy?

Reply:
Indeed! Fixed that now.